src/app/pages/components/tree/input.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+    selector: 'ngx-modal',
+    templateUrl: './input.component.html',
+})
+export class InputComponent implements OnInit {
+
+    public form: FormGroup;
+    modalHeader: string;
+    modalOperation: number;
+    modalNode: any;
+    modalTree: any;
+    modalTreeData: any;
+
+
+
+    constructor(private activeModal: NgbActiveModal,
+        private fb: FormBuilder,
+    ) { }
+
+    ngOnInit() {
+        this.form = this.fb.group({
+            node: [null, Validators.compose([Validators.required])],
+        });
+        if (this.modalOperation === 1) {
+            this.form.patchValue({
+                node: this.modalNode.data.name,
+            });
+        }
+    }
+
+    closeModal() {
```

Comment:
Ive seen in a few places in the repo that they type void return types as well
`closeModal(): void {`

This would apply to all methods in your PR